src/logging/http-exceptions-logger.filter.ts

```diff
@@ -12,6 +12,38 @@ import { formatISO } from 'date-fns';
 export class AllExceptionsFilterLogger implements ExceptionFilter {
   constructor(private readonly logger: Logger) {}
 
+  generateErrorBasedOnCurrentEnvironment(
```

Comment:
- Add **generateErrorBasedOnCurrentEnvironment** function in a separate file. Also make it pure function, by passing NODE_ENV as a variable. Making it pure function will make test much easier.
- Add tests for the function